app/models/report.rb

```diff
@@ -4,24 +4,44 @@ class Report
   attr_reader :exercise_logs, :pain_logs, :pt_sessions, :pains, :exercises, :body_parts
 
   def initialize(args)
-    @filter_params = args
-    @exercise_logs ||= query_logs('exercise_logs')
-    @pain_logs ||= query_logs('pain_logs')
-    @pt_sessions ||= query_logs('pt_sessions')
-    @pains ||= @filter_params[:user].pains
-    @exercises ||= @filter_params[:user].exercises
-    @body_parts ||= @filter_params[:user].body_parts
+    @exercise_logs = args[:exercise_logs]
+    @pain_logs = args[:pain_logs]
+    @pt_sessions = args[:pt_sessions]
+    @pains = args[:pains]
+    @exercises = args[:exercises]
+    @body_parts = args[:body_parts]
   end
 
-  private
+  class << self
+    def build_report(filter_params) # rubocop:disable Metrics/MethodLength
+      @filter_params = filter_params
+      exercise_logs = query_logs('exercise_logs')
+      pain_logs = query_logs('pain_logs')
+      pt_sessions = query_logs('pt_sessions')
+      pains = filter_params[:user].pains
+      exercises = filter_params[:user].exercises
+      body_parts = filter_params[:user].body_parts
 
-  def query_logs(log_type)
-    logs = log_type.classify
-                   .constantize
-                   .where(user_id: @filter_params[:user].id)
+      Report.new(
+        exercise_logs: exercise_logs,
+        pain_logs: pain_logs,
+        pt_sessions: pt_sessions,
+        pains: pains,
+        exercises: exercises,
+        body_parts: body_parts
+      )
+    end
 
-    logs = logs.for_body_part(@filter_params[:body_part_id]) if @filter_params[:body_part_id].present?
-    logs = logs.for_past_n_days(@filter_params[:timeframe].to_i) if @filter_params[:timeframe].present?
-    logs
+    private
+
+    def query_logs(log_type)
+      logs = log_type.classify
```

Comment:
Since you aren't using user input or something here you could probably change the `query_logs` method to accept a `log_class` like `ExerciseLog` and ditch the `classify` and `constantize` calls.

Reply:
Oh yeah. 🤦‍♀ Good idea!